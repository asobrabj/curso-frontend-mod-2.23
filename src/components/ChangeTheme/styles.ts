import styled from 'styled-components'

export const ContainerTheme = styled.div`
  cursor: pointer;
  margin-left: 450px;
  margin-top: 5px;
  margin-right: 80px;
  position: relative;
  border: 1px solid ${(props) => props.theme.color.shadow};
  width: 32px;
  height: 32px;
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;

  > i {
    position: absolute;
    top: 6px;
    left: 6px;
    font-size: 18px;
    border-radius: 50%;
    color: ${(props) => props.theme.color.hardShadow};
  }

  @media (max-width: 820px) {
    margin-left: 226px;
  }
`
