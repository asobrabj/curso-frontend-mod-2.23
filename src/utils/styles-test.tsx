import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import theme from '../styles/darkTheme'

export const renderTheme = (Children: React.ReactNode) => {
  return render(<ThemeProvider theme={theme}> {Children} </ThemeProvider>)
}
