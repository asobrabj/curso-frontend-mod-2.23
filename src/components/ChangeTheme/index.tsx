import { IoMoon, IoSunnySharp } from 'react-icons/io5'
import * as S from './styles'

interface ITheme {
  isDark: boolean
  changeTheme: () => void
}

export default function ChangeTheme({ changeTheme, isDark }: ITheme) {
  return (
    <S.ContainerTheme
      onClick={changeTheme}
      title="change theme"
      data-testid="change theme"
    >
      <i title="icon theme">{isDark ? <IoSunnySharp /> : <IoMoon />}</i>
    </S.ContainerTheme>
  )
}
