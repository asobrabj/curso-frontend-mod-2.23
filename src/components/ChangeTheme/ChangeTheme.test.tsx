import { screen } from '@testing-library/react'
import { renderTheme } from '../../utils/styles-test'
import ChangeTheme from '.'

describe('ChangeTheme', () => {
  it('deve renderizar o componente corretamente', () => {
    renderTheme(<ChangeTheme isDark={false} changeTheme={() => {}} />)
    const div = screen.getByTitle(/change theme/i)
    expect(div).toBeInTheDocument()
  })

  it('deve renderizar o componente corretamente', () => {
    renderTheme(<ChangeTheme isDark={false} changeTheme={() => {}} />)
    const div = screen.getByTitle(/icon theme/i)
    expect(div).toBeInTheDocument()
  })

  it('snapshot', () => {
    const { container } = renderTheme(
      <ChangeTheme isDark={false} changeTheme={() => {}} />
    )
    expect(container).toMatchSnapshot()
  })
})
