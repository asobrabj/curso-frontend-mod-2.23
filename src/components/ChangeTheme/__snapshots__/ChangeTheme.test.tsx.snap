// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangeTheme snapshot 1`] = `
.c0 {
  cursor: pointer;
  margin-left: 450px;
  margin-top: 5px;
  margin-right: 80px;
  position: relative;
  border: 1px solid #999;
  width: 32px;
  height: 32px;
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
}

.c0 >i {
  position: absolute;
  top: 6px;
  left: 6px;
  font-size: 18px;
  border-radius: 50%;
  color: #bdbbbb;
}

@media (max-width: 820px) {
  .c0 {
    margin-left: 226px;
  }
}

<div>
   
  <div
    class="c0"
    data-testid="change theme"
    title="change theme"
  >
    <i
      title="icon theme"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M264 480A232 232 0 0132 248c0-94 54-178.28 137.61-214.67a16 16 0 0121.06 21.06C181.07 76.43 176 104.66 176 136c0 110.28 89.72 200 200 200 31.34 0 59.57-5.07 81.61-14.67a16 16 0 0121.06 21.06C442.28 426 358 480 264 480z"
        />
      </svg>
    </i>
  </div>
   
</div>
`;
